components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -112,7 +112,9 @@ class BrowserIcons(
         }
 
         // (3) Finally process the icon.
-        return process(context, processors, request, resource, icon)
+        return process(context, processors, request, resource, icon).run {
+            if (bitmap != null) this else generator.generate(context, request)
+        }
```

Comment:
Did you see the bitmap being null here? Just curious since the previous code should already have returned an `Icon` with bitmap.

Reply:
Looking more at it almost looks like we can make `Icon.bitmap` not nullable.

Reply:
`ResizingProcessor` returns a null bitmap when the icon is too small to indicate that a generated icon should be used instead. We can opt to have a special variant of `Icon` with a nullable bitmap or let process return a `Icon?`.

Reply:
After playing around with it, I think letting process return an `Icon?` is the best way to do this. I updated `Icon.bitmap` to be not-null.